src/appleseed/renderer/kernel/rendering/final/adaptivetilerenderer.cpp

```diff
@@ -552,7 +552,7 @@ namespace
                         {
                             Color3f variation;
                             m_variation_aov_tile->get_pixel(pt.x, pt.y, variation);
-                            variation[0] += pb.m_block_error;
+                            variation[0] += pb.m_block_error / m_params.m_noise_threshold;
```

Comment:
What if the noise threshold is zero ?

Reply:
The noise threshold has a minimum value of 0.0001 . It cannot be set to 0 either in the .appleseed file or through appleseed studio.
https://github.com/appleseedhq/appleseed/blob/127bbc39b2bd9a4cbcaff2b94552757aec6f8058/src/appleseed/renderer/kernel/rendering/final/adaptivetilerenderer.cpp#L1014 ensures it I think.

Reply:
Ok nice

Reply:
Ok nice

Reply:
Could you share some images of what this change does ?

Reply:
![Screenshot from 2019-04-05 07-16-09](https://user-images.githubusercontent.com/25225896/55641789-5ca92c80-57ec-11e9-831e-6624b88df165.png)
The scene 
![NormBuild](https://user-images.githubusercontent.com/25225896/55641791-60d54a00-57ec-11e9-9cf5-69ce312004cd.png)
Using thresholds of 0.004, 0.3, 0.6
![NormalWithoutPost](https://user-images.githubusercontent.com/25225896/55641797-65016780-57ec-11e9-8492-b0818990d645.png)
Without the post processing stage

The problem with this is that after the post processing stage, the aov image with my changes on and those without are exactly the same. I am trying to figure out what is causing the problem

Reply:
ok. I think i already tried this when I first implented it and it wasnt that great. Though I was normalizing while rendering whereas here you normalize it only for the final aov.